app/models/manageiq/providers/hawkular/middleware_manager/middleware_diagnostic_report.rb

```diff
@@ -0,0 +1,139 @@
+require 'timeout'
+
+module ManageIQ::Providers
+  class Hawkular::MiddlewareManager::MiddlewareDiagnosticReport < ApplicationRecord
+    self.table_name = 'middleware_diagnostic_reports'
+
+    STATUS_QUEUED = 'Queued'.freeze
+    STATUS_RUNNING = 'Running'.freeze
+    STATUS_ERROR = 'Error'.freeze
+    STATUS_READY = 'Ready'.freeze
+
+    belongs_to :middleware_server
+    has_one :binary_blob, :as => :resource, :dependent => :destroy
+
+    validates :middleware_server_id, :requesting_user, :presence => true
+    validates :status, :inclusion => { :in => [STATUS_QUEUED, STATUS_RUNNING, STATUS_ERROR, STATUS_READY] }
+
+    delegate :ext_management_system, :to => :middleware_server
+    delegate :ems_id, :to => :middleware_server
+    after_create :enqueue_job
+    before_create :set_queued_date
+
+    after_initialize do |item|
+      item.status = STATUS_QUEUED if new_record?
+    end
+
+    def queued?
```

Comment:
We could use [rails enums](http://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html) to simplify the logic.